### DISCUSSION TITLE
Can I change the status of a previous job?

### DISCUSSION THREAD

**Discussion by PaulLesur** [community/community#26413] [Actions]
<p>Hi,<br>
I was wondering if I could change the status of a previous job in my workflow from another job.<br>
In fact I have a workflow which runs on pull requests, in this workflow I have a matrix job which splits my tests suites between 20 runners in parallel. After this matrix job I have a “Retry job” which rerun the failed jobs. If the tests passed on rerun, then I want to set the status of runners to success to not “disturb” the combined status of the PR.<br>
So I was searching for a way to set status of previous job or to remove a status from a commit.</p>
<p>Thank you !</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/paullesur">@PaulLesur</a>,</p>
<p>Currently, we seem have not methods to re-run an individual job in a workflow run. We only can re-run the whole workflow run.</p>
<p>If your projects really need this feature, I recommend you directly share a feature request in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="noopener nofollow ugc"><strong>Feedback form for GitHub Actions</strong></a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by PaulLesur** [comment]
<p>Hi, thank you for your answer !</p>
<p>In the meantime I achieved something using Checks API. In my last job I take the run ID of a previous job and set it to <em>success</em>, it works BUT at the end of the workflow, the status comes back to <em>failure</em>… I can see the icon going from red cross, to green tick and to red cross again…</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/paullesur">@PaulLesur</a>,</p>
<p>The <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/checks" rel="noopener nofollow ugc"><strong>Checks API</strong></a> is applied to the checks created by the <strong>Checks API</strong> via the GitHub Apps.<br>
For example, if you use “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#create-a-check-run" rel="noopener nofollow ugc"><strong>Create a check run</strong></a>” endpoint to create a check run with a specified status, then you can use the “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#update-a-check-run" rel="noopener nofollow ugc"><strong>Update a check run</strong></a>” endpoint to change the status.</p>
<p>The “<strong>Update a check run</strong>” endpoint is not functionally equivalent to re-running a job.</p>

**Comment by PaulLesur** [comment]
<p>Thank you,<br>
My solution is to trigger another workflow wich update checks of my PR, but this is not a great solution…</p>
